### ISSUE TITLE
[no-unsafe-mutable-readonly-assignment] incorrectly considers const array to be mutable

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#96]
For some reason the follow is considered mutable -> readonly assignment (but shouldn't be):

```ts
const foo: ReadonlyArray<string> = [] as const;
```

**Comment by danielnixon** [comment]
The length property of tuples is technically mutable while it is readonly in `ReadonlyArray`. So assigning a tuple to a readonly array is technically assigning mutable to readonly. I've added a special case to ignore length.
